src/re2parser.cc

```diff
@@ -560,18 +560,26 @@ namespace {
  * @param pattern regex as string
  * @param use_epsilon whether to create NFA with epsilon transitions or not
  * @param epsilon_value value, that will represent epsilon on transitions
+ * @param use_reduce if set to true the result is trimmed and reduced using simulation reduction
  * @return mata::Nfa::Nfa corresponding to pattern
  */
-void Mata::RE2Parser::create_nfa(Nfa::Nfa* nfa, const std::string& pattern, bool use_epsilon, int epsilon_value) {
+void Mata::RE2Parser::create_nfa(Nfa::Nfa* nfa, const std::string& pattern, bool use_epsilon, int epsilon_value, bool use_reduce) {
     if (nfa == NULL) {
         throw std::runtime_error("create_nfa: nfa should not be NULL");
     }
 
     RegexParser regexParser{};
     auto parsed_regex = regexParser.parse_regex_string(pattern);
     auto program = parsed_regex->CompileToProg(regexParser.options.max_mem() * 2 / 3);
-    regexParser.convert_pro_to_nfa(nfa, program, use_epsilon, epsilon_value);
+    regexParser.convert_pro_to_nfa(nfa, program, true, epsilon_value);
```

Comment:
Is this change intended? Seems quite out of place wrt rest of the changes.

Reply:
It is. The workaround is to always use epsilon (hence the change from `use_epsilon` to `true`) and remove epsilon symbols afterwards, if required.

Reply:
Yes, I changed it deliberately. It seems that inside `convert_pro_to_nfa` the code removing epsilons is not working properly (if `use_epsilon` is set to true, the problematic code is not used).

Reply:
I suppose we remove the problematic code from `convert_pro_to_nfa` entirely at some point. So this parameter will not be used anymore